official/vision/beta/projects/panoptic_maskrcnn/modeling/panoptic_maskrcnn_model.py

```diff
@@ -0,0 +1,417 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Mask R-CNN model."""
+
+from typing import Any, List, Mapping, Optional, Union
+
+# Import libraries
+import tensorflow as tf
+
+from official.vision.beta.ops import anchor
+from official.vision.beta.ops import box_ops
+
+
+@tf.keras.utils.register_keras_serializable(package='Vision')
+class PanopticMaskRCNNModel(tf.keras.Model):

```

Comment:
Can you subclass MaskRCNNModel model instead? I think you can reuse the init, and call functions, and just add the segmentation, and panoptic layer on top.